### PR TITLE
Add remote signing diagram

### PR DESCRIPTION
**Pull request by OR13**
Screenshot 2024-01-18 at 6 30 50 PM

### CODE REVIEW

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -1293,3 +1294,64 @@ transparent-statement:sha-256:base64url:5i6UeRzg1...qnGmr1o
 data:application/cose;base64,SGVsb...xkIQ==
 ~~~
 {: #example-transparent-statement-data-url align="left" title="Example Transparent Statement Data URL"}
+
+# Signing Statements Remotely
+
+Some statements are too large, or sensitive to send over a network to a remote signer.
```
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:1300]:
```suggestion
Some statements are too large or too sensitive to send over a network to a remote signer.
```

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -1293,3 +1294,64 @@ transparent-statement:sha-256:base64url:5i6UeRzg1...qnGmr1o
...
+
+Recall that a Statement is an artifact, structured data regarding an artifact, or a hash.
+
+A Statement becomes a payload, which becomes to-be-signed bytes.
+
+A Signed Statement (cose-sign1) MUST be produced from the to-be-signed bytes according to {{Section 4.4 of RFC9052}}.
```
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:1306]:
```suggestion
Some statements, such as digital artifacts structured data regarding an artifacts, are too large or too sensitive to be send to a remote Transparency Services over the Internet.
In these cases a statement can also be hash, which becomes the payload included in COSE to-be-signed bytes.
A Signed Statement (cose-sign1) MUST be produced from the to-be-signed bytes according to {{Section 4.4 of RFC9052}}.
```

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -187,6 +187,8 @@ The terms defined in this section have special meaning in the context of Supply
 When used in text, the corresponding terms are capitalized.
 To ensure readability, only a core set of terms is included in this section.
 
+The terms "header", "payload", and "to-be-signed bytes" are defined in {{RFC9052}}.
```
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:190]: :+1:

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:1306]: Wait, how is this PR related to #155?

**Review by henkbirkholz — APPROVED** [review]
lgtm

**Review by aj-stein-nist — APPROVED** [review]

File: draft-ietf-scitt-architecture.md [hunk 4]
```diff
@@ -1293,3 +1294,60 @@ transparent-statement:sha-256:base64url:5i6UeRzg1...qnGmr1o
...
+      |                     .----+-----.
+      |                    |   Header   |
+      |                     '----+-----'
+      |                          |   |
+      v                          v   v
+    .-+-----------.     .--------+---+---.
+   /             /     /                  \
+  /    Sign     +<----+ To Be Signed Bytes |
+ /             /       \                  /
+'-------------'         '----------------'
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:1346]:
```suggestion
'-----+-------'         '----------------'
```

Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:1346]: Minor fix to differentiate lines "behind", vs. from. 
In this case, I believe @OR13 is showing the COSE Sign 1 Document comes from the Sign process.

File: draft-ietf-scitt-architecture.md [hunk 5]
```diff
@@ -1293,3 +1294,60 @@ transparent-statement:sha-256:base64url:5i6UeRzg1...qnGmr1o
...
+
+           ...  Producer Network ...
+
+                      ...
+
+           ...   Issuer Network ...
+
+                                     |
++-----------+                        |
+| Identity  |     (iss, x5t)         |
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:1329]: As a "document", should this have rounded corners?
I have time this afternoon to reivew and make PRs to tweak before approving.

**Comment by OR13** [comment]
The diagram reflects my intention.

File: draft-ietf-scitt-architecture.md (hunk 4 above)
Comment (OR13) [reply draft-ietf-scitt-architecture.md:1346]: The lines are a correct as they are.

File: draft-ietf-scitt-architecture.md (hunk 5 above)
Comment (OR13) [reply draft-ietf-scitt-architecture.md:1329]: This box represents a concept for an identity document, not a concrete identity document such as a certificate.

I don't intend to update the PR, but you are welcome to.

Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:1329]: As the Identity Document pre-exists outside of that data flow, I'd be okay with keeping that box different.

File: draft-ietf-scitt-architecture.md [hunk 6]
```diff
@@ -1293,3 +1294,60 @@ transparent-statement:sha-256:base64url:5i6UeRzg1...qnGmr1o
...
+| Document  +--------------------+   |
++-----+-----+                    |   |
+      ^                          |   |
+      |                          |   |
+ .----+-------.                  |   |
+| Private Key  |                 |   |
+ '----+-------'                  v   |
+      |                     .----+-----.
+      |                    |   Header   |
```
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:1338]: The Header part and the "To be Signed bytes are mis-leading to me"

Comment (yogeshbdeshpande) [reply draft-ietf-scitt-architecture.md:1338]: Specially the `Payload` sections `Arrow` enters into the Header section. What are we trying to depict?

Comment (yogeshbdeshpande) [reply draft-ietf-scitt-architecture.md:1338]: Also if I am not mis-taken `To be signed bytes` in our case is actually the `payload` ?

If it is, then can we keep the two same so as to logically link the two sections of the diagrams without physically linking them?

**Comment by SteveLasker** [comment]
Here's a slightly modified view to remove the lines behind, add arrows consistenty
image

**Review by SteveLasker — CHANGES_REQUESTED** [review]
I've added #173 to cleanup a few elements

**Review by SteveLasker — APPROVED** [review]

**Pull request merged**